### PR TITLE
More generic docs for use in APM Server

### PR DESCRIPTION
**Pull request by roncohen**
* Use specific beat name name instead of "the Beat"
* Make Logstash and other outputs optional for the beat
* Decouple the index pattern from the beat name

### CODE REVIEW

File: libbeat/docs/dashboardsconfig.asciidoc [hunk 1]
```diff
@@ -20,7 +20,7 @@ To load the dashboards, you can either enable dashboard loading in the
 run the `setup` command. Dashboard loading is disabled by default.
 
 When dashboard loading is enabled, {beatname_uc} uses the Kibana API to load the
-sample dashboards. Dashboard loading is only attempted at Beat startup.
+sample dashboards. Dashboard loading is only attempted when {beatname_uc} startups up.
```
Comment (dedemorton) [inline libbeat/docs/dashboardsconfig.asciidoc]: typo: startups up

**Comment by ruflin** [comment]
@roncohen Seeing you pushed some more change, I'm curious if it's ready for merge? As Dede is off this week, please ping me if yes.

**Comment by roncohen** [comment]
there's more incoming @ruflin

**Comment by roncohen** [comment]
quite a few things changed so this probably needs another round from @dedemorton

File: libbeat/docs/loggingconfig.asciidoc [hunk 2]
```diff
@@ -63,11 +63,11 @@ errors, there will be no log file in the directory specified for logs.
 [[level]]
 ==== `logging.level`
 
-Minimum log level. One of `debug`, `info`, `warning`, or `error`. The default
+Minimum log level. One of `debug`, `info`, `warn`, or `error`. The default
```
Comment (tsg) [inline libbeat/docs/loggingconfig.asciidoc]: Remember to change this back to `warning`, as we went for making `warning` work again in https://github.com/elastic/beats/pull/6240

Comment (roncohen) [reply libbeat/docs/loggingconfig.asciidoc]: fixed. Thanks for the reminder.

**Comment by graphaelli** [comment]
@tsg @dedemorton can one or both of you sign off this one?

**Comment by dedemorton** [comment]
@graphaelli Oh sorry. I was on vacation when the notification came through and didn't see the notification from GitHub. I'll take a look sometime today.

File: libbeat/docs/outputconfig.asciidoc [hunk 3]
```diff
@@ -13,6 +13,12 @@
 [[configuring-output]]
 == Configure the output
 
+ifdef::only-elasticsearch[]
+You configure {beatname_uc} to write to Elasticsearch by setting options in
+the `output.elasticsearch` of the +{beatname_lc}.yml+ config file.
```
Comment (dedemorton) [inline libbeat/docs/outputconfig.asciidoc]: I'd say either "in the `output.elasticsearch` section of the +{beatname_lc}.yml+ config file" or "under `output.elasticsearch` in the +{beatname_lc}.yml+ config file."

File: libbeat/docs/shared-ssl-config.asciidoc [hunk 4]
```diff
@@ -1,5 +1,5 @@
 [[configuration-ssl]]
-== Specify SSL settings
+== SSL settings for outputs
```
Comment (dedemorton) [inline libbeat/docs/shared-ssl-config.asciidoc]: The SSL settings are relevant for other stuff (like kibana), too, so I wouldn't change the title here.

File: libbeat/docs/template-config.asciidoc [hunk 5]
```diff
@@ -8,18 +8,22 @@ mappings in Elasticsearch. If template loading is enabled (the default),
...
+
 You can adjust the following settings to load your own template or overwrite an
 existing one.
 
 *`setup.template.enabled`*:: Set to false to disable template loading. If set this to false,
 you must <<load-template-manually,load the template manually>>.
 
 *`setup.template.name`*:: The name of the template. The default is
-+{beatname_lc}+. The Beat version is always appended to the given
++{beatname_lc}-*+. The {beatname_uc} version is always appended to the given
```
Comment (dedemorton) [inline libbeat/docs/template-config.asciidoc]: Does this asterisk really belong here?

**Comment by roncohen** [comment]
I've addressed your comments in 5a4de2b. Let me know if there's anything else

**Comment by roncohen** [comment]
@dedemorton OK to merge?

**Comment by dedemorton** [comment]
@roncohen Yes (as long as the CI failures look OK to you...TBH, I'm never quite sure which CI failures I can ignore).

Do we need to backport these changes to 6.2? Or are they for 6.3 and later?

**Pull request merged**